"Expected NaN spaces" indent error with multiline string and arguments: 'off'

**Tell us about your environment**

* **ESLint Version:** 4.4.0
* **Node Version:** 6.10.3
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    es6: true,
  },
  rules: {
    indent: ['error', 2, {
      CallExpression: {
        arguments: 'off',
      },
    }],
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**

This is a reduced test case showing the issue:
```js
a(`b
c`, function() {
  d;
});
```

**What did you expect to happen?**

Ideally no error, but if there is an error, it should be sensible.

**What actually happened? Please include the actual, raw output from ESLint.**
eslint gives this error:
```
3:1  error  Expected indentation of NaN spaces but found 2  indent
```

Running `eslint --fix` treats NaN as 0 and removes all indentation from all affected lines.

Changing the multiline string to a single-line string fixes the issue, and removing `arguments: 'off'` in the config also fixes the issue.

**Context:**
This type of code came up when converting multiline strings from CoffeeScript ([example](http://decaffeinate-project.org/repl/#?evaluate=true&stage=full&code=it%20'this%20is%20a%20multiline%0A%20%20%20%20test%20name'%2C%20-%3E%0A%20%20expect(1).eql%201%0A%20%20return)), and it appears a number of times in my codebase. A workaround is to rewrite the multiline strings as single-line strings, but certainly would be nice for this to be fixed within eslint.

Another example with the same config:

```js
a(
  new B({
    c,
  })
);
```

Gives this error:
```
  3:1  error  Expected indentation of NaN spaces but found 4  indent
  4:1  error  Expected indentation of NaN spaces but found 2  indent
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -783,7 +783,7 @@ module.exports = {
             offsets.setDesiredOffsets(
                 [startToken.range[1], endToken.range[0]],
                 startToken,
-                offset === "first" ? 1 : offset
+                typeof offset === "number" ? offset : 1
             );
             offsets.setDesiredOffset(endToken, startToken, 0);
 

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -4915,6 +4915,24 @@ ruleTester.run("indent", rule, {
                     );
                 }
             `
+        },
+        {
+            code: unIndent`
+                a(b
+                  , c
+                )
+            `,
+            options: [2, { CallExpression: { arguments: "off" } }]
+        },
+        {
+            code: unIndent`
+                a(
+                  new B({
+                    c,
+                  })
+                );
+            `,
+            options: [2, { CallExpression: { arguments: "off" } }]
         }
     ],
 

